appscale/tools/remote_helper.py

```diff
@@ -108,6 +108,31 @@ def start_all_nodes(cls, options, count):
 
     return instance_ids, public_ips, private_ips
 
+  @classmethod
+  def enable_head_node_root_ssh(cls, options, head_node):
+    """Enables root logins on the head node, enables SSH access,
+    and copies the user's SSH key to the head node.
+
+    Args:
+      options: A Namespace that specifies the cloud infrastructure to use, as
+        well as how to interact with that cloud.
+      head_node: The head_node from the NodeLayout.
+    """
+    cls.sleep_until_port_is_open(head_node.public_ip, cls.SSH_PORT,
+                                 options.verbose)
+
+    # Since GCE v1beta15, SSH keys don't immediately get injected to newly
+    # spawned VMs. It takes around 30 seconds, so sleep a bit longer to be
+    # sure.
+    if options.infrastructure == 'gce':
```

Comment:
This check is irrelevant in this context. I think it should be done before this method.